### DISCUSSION TITLE
How to set cube-face camera for probe generation?

### DISCUSSION THREAD

**Discussion by Honya2000** [google/filament#4788] [General]
Is there some function in the code I can use to setup the camera for specific IBL Probe face (to render HDR scene for each direction)?
I want to generate/cache IBL Prefiltered cubemap in my scene, but don't kow exactly how to set the camera for each face.
Description of IBL cube-maps is quite confusing about mirroring settings, etc...

**Comment by ThomasGorisse** [comment]
You can set your own image for each Cubemap face like this (in kotlin):
```kotlin
val syboxCubemap = Texture.Builder()
    .width(width)
    .height(height)
    .levels(0xff)
    .sampler(Texture.Sampler.SAMPLER_CUBEMAP)
    .build(Filament.engine)
    .apply {
        // Set each face image from your hdr or distinct images here
        setImage(Filament.engine, 0, imageBuffer, faceOffsets)
    }
```

and then create the Skybox and apply it to the Scene:
```kotlin
scene.skybox = Skybox.Builder()
    .environment(syboxCubemap)
    .build(Filament.engine)
```

*Note that's it a runtime solution but you can also pre-compile a environment_skybox.ktx and environment_ibl.ktx from your modified hdr file using cmgen*

**Reply by Honya2000** [reply]
I'm asking about something different.
I know how to upload IBL.

But I want to generate and precache local probes inside real scene.
For generation purpose I have to setup the camera for each face from specific probe position and render the scene.
So I asking how to setup the camera for each cube-map face.

**Reply by Honya2000** [reply]
Cmgen doesn't use real scene. It also generate cubemaps from existing images.
So it kinda useless for my needs...

**Comment by Honya2000** [comment]
Bump.
Nobody knows how to setup the camera?
